GOSR2.py

```diff
@@ -0,0 +1,200 @@
+import os
+
+from Bio import SeqIO
+from Bio import SeqUtils
+from Bio import SeqRecord
+from abc import ABC, abstractmethod
+
+
+def filter_fastq(input_path: str, output_filename: str = '', gc_bounds: tuple = (0, 100), length_bounds: tuple = (0, 2**32), quality_threshold: int = 0)->list:
+    """
+    Filter fastq-sequences based on its gc-content, length and quality.
+    arguments:
+        - input_path (str): path to the fastq-file for reading
+        - output_filename (str): path for saving file with result (filtered fastq)
+        - gc_bounds (tuple): lower and upper borders for gc-content of sequence in percent
+        - length_bounds (tuple): lower and upper borders for length of sequence
+        - quality_threshold (int): lower border for quality of sequence
+    return:
+        - list of SeqRecords: filtered fastq-sequences
+    """
+    
+    if not isinstance(gc_bounds, tuple):
+        gc_bounds = tuple((0, int(gc_bounds)))
+    if not isinstance(length_bounds, tuple):
+        length_bounds = tuple((0, int(length_bounds)))
+
+    suitable_seqs = list(
+        record
+        for record in SeqIO.parse(input_path, "fastq")
+        if ((gc_bounds[0] <= SeqUtils.gc_fraction(record.seq)*100 <= gc_bounds[1]) and 
+            (length_bounds[0] <= len(record) <= length_bounds[1]) and 
+            (sum(record.letter_annotations['phred_quality']) / len(record) >= quality_threshold))
+    )  
+
+    if output_filename == '':
+        output_filename = 'good_quality.fastq'
+    dir_name = 'fastq_filtrator_results'
+    if not output_filename[-6:] == '.fastq':
+        output_filename = output_filename + '.fastq'
+    if not os.path.isdir(dir_name):
+        os.mkdir(dir_name)
+    output_path = os.path.join(dir_name, output_filename)
+    if os.path.isfile(output_path):
+        raise ValueError('File with such name is exist. Please, use another name for your output file.')
+    SeqIO.write(suitable_seqs, output_path, 'fastq')
+    print('The result of fastq-filtering is saved in', output_path)
+    
+    return suitable_seqs
+
+
+class BiologicalSequence(ABC, str):
+    """
+    Abstract class for NucleicAcidSequence and AminoAcidSequence. 
+    attributes:
+        - seq (str): sequence
+    methods:
+        - check_alphabet: check whether the seq matches the right biological alphabet
+    """
+
+    def __init__(self, seq: str):

```

Comment:
```suggestion
   @abstractmethod
    def __init__(self, seq: str):

```
Абстрактному классу - абстрактыне методы :)

Reply:
Обрати внимание - у тебя маркдаун распознал `__` как обозначение жирного текста. Чтобы было как ты хотела - надо ставить штрихи с двух сторон (они на букве ё) - так будет отображено как код

Reply:
И да, луше ничего не реализовывать в абстрактных классах, верно